### PR TITLE
Benchmark snakemake 2

### PR DESCRIPTION
**Pull request by schoersch**

### CODE REVIEW

**Comment by schoersch** [comment]
@smilesun this is the snakemake benchmark PR without the boxplots.

As discussed already, the code analysis shows the two usages of `eval()` as issues, which are used to allow expressions for the hyperparameter sampling in the config yaml

**Review by smilesun — COMMENTED** [review]
do we need the aggregated results file?

**Comment by schoersch** [comment]
> do we need the aggregated results file?

You mean in ztest_files `aggret_rest_test1` and 2? These two files are loaded by @Car-la-F in the tests to test the plot generation.

**Comment by smilesun** [comment]
@schoersch @Car-la-F ,  after trying for 1 hour, I found the argument batchsize does not exist in DomainLab, the correct argumetn should be "bs", now the demo benchmakr works on my laptop, will push the changes.

**Comment by smilesun** [comment]
@Car-la-F , I am editing many places in the branch, make sure to "pull" every now and then so you wont encounter a conflict, but otherwise just resovle the conflict

**Comment by smilesun** [comment]
@Car-la-F , did I break anything? 
```
FAILED tests/test_benchmark_plots.py::test_benchm_plots - FileNotFoundError: [Errno 2] No such file or directory: 'data/ztest_files/aggret_res_test3'
FAILED tests/test_run_experiment.py::test_run_experiment - KeyError: 'allocated_bytes.all.current'
```

**Comment by smilesun** [comment]
> @Car-la-F , did I break anything?
> 
> ```
> FAILED tests/test_benchmark_plots.py::test_benchm_plots - FileNotFoundError: [Errno 2] No such file or directory: 'data/ztest_files/aggret_res_test3'
> FAILED tests/test_run_experiment.py::test_run_experiment - KeyError: 'allocated_bytes.all.current'
> ```

sorry, just saw, it is fixed already

**Comment by schoersch** [comment]
@smilesun I found a bug in the run_experiment and fixed it there. However, updating I noticed that the benchmark.smk aborts with syntax errors.

**Comment by smilesun** [comment]
@schoersch , SORRY i broke your hyper-parmaeter sampling code, but will fix it

**Comment by smilesun** [comment]
@schoersch @Car-la-F , thanks for the hardwork! very impressive and high quality code!

Now the branch is mergable, and I will do after a final test on my local machine with running the benchmark. 

I could not get the literal_eval work on the hyper-parameter sampling, that is why codacy is failing, but i think we can first live with it.

**Review by smilesun — APPROVED** [review]

File: domainlab/utils/aggret_res_test1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+param_index, task, algo, epos, te_d, seed, params, acc, precision, recall, specificity, f1, auroc
```
[the remark this answers is not in the document]
Comment (Car-la-F) [reply domainlab/utils/aggret_res_test1]: @smilesun,
I do agree, this file should not be in the domainlab folder.

But from my understanding this file is saved as `data/ztest_files/aggret_res_test*`, that's also where I do find it in the master.  I couln't find a replicate in the `domainlab/utils` folder though, I guess you removed this replicate already. 

The important point is, that only the files in `data/ztest_files` are needed for the tests.

**Comment by schoersch** [comment]
> @schoersch @Car-la-F , thanks for the hardwork! very impressive and high quality code!
> 
> Now the branch is mergable, and I will do after a final test on my local machine with running the benchmark.
> 
> I could not get the literal_eval work on the hyper-parameter sampling, that is why codacy is failing, but i think we can first live with it.

From the [docs](Evaluate an expression node or a string containing only a Python literal or container display. The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis.):
_Evaluate an expression node or a string containing only a Python literal or container display. The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis._

I.e. `literal_eval` does not allow expressions like `a < b`. Thus we have to use the more open `eval` here.

File: domainlab/utils/aggret_res_test1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (smilesun) [reply domainlab/utils/aggret_res_test1]: I have moved them to the data folder alreadyy

**Comment by smilesun** [comment]
> > @schoersch @Car-la-F , thanks for the hardwork! very impressive and high quality code!
> > Now the branch is mergable, and I will do after a final test on my local machine with running the benchmark.
> > I could not get the literal_eval work on the hyper-parameter sampling, that is why codacy is failing, but i think we can first live with it.
> 
> From the [docs](Evaluate an expression node or a string containing only a Python literal or container display. The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis.): _Evaluate an expression node or a string containing only a Python literal or container display. The string or node provided may only consist of the following Python literal structures: strings, bytes, numbers, tuples, lists, dicts, sets, booleans, None and Ellipsis._
> 
> I.e. `literal_eval` does not allow expressions like `a < b`. Thus we have to use the more open `eval` here.
@schoersch 
literal_eval actually fails at evaluating expression "p_3", which is some float number

**Pull request merged**